src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/Events.ts

```diff
@@ -0,0 +1,40 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as psList from 'ps-list';
+import { DebugSession } from 'vscode';
+
+import { RazorLogger } from '../RazorLogger';
+
+export async function onDidTerminateDebugSession(
+  event: DebugSession,
+  logger: RazorLogger,
+) {
+  logger.logVerbose('Terminating debugging session...');
+  if (process.platform !== 'win32') {
+    try {
+      const processes = await psList();
+      const devserver = processes.find(
```

Comment:
Would it make more sense to keep a global variable with the started dev server (or servers if we support more than one)?

Reply:
Moving this comment to the right thread.

I did this and moved the `onTerminateDebugSession` handler to inside the configuration provider.

```js
const app = spawn("dotnet", ["run"]);
this.vscodeType.debug.onDidTerminateDebugSession(event => onDidTerminateDebugSession(event, this.logger, app));
```

Is registering event handlers for the debugger in the config provider like this a sensible pattern?